types/indoorMapFloorOutlines.ts

```diff
@@ -0,0 +1,39 @@
+import type { EventHandler } from "./event";
+import type Map from "./map";
```

Comment:
Exposing my lack of typescript knowledge - is there any particular reason why we `import type` here but in the `index.d.ts` we just `import` ?

Reply:
Importing types means that the thing that we are importing won't be used at runtime, so it will be discarded at compile time making sure we're not importing anything that might cause side effects.

In `types/indoorMapFloorOutlines.ts` we're importing just the type because we are not using map at runtime, while in the `index.d.ts` we're importing it, to then re-export a `Map` namespace/interface that can be used at runtime.

Note that TypeScript definitions are stripped at compile time, and what the browser runs is plain old non-typed JavaScript.